exist-core/src/main/java/org/exist/xquery/functions/array/ArrayType.java

```diff
@@ -382,6 +382,30 @@ public static Sequence flatten(Sequence input) throws XPathException {
         return flatten ? flatten(input, new ValueSequence(input.getItemCount() * 2)) : input;
     }
 
+    @Override
+    public boolean containsReference(final Item item) {
+        for (int i = 0; i < vector.length(); i++) {
+            final Sequence value = vector.nth(i);
+            if (value == item) {
+                return true;
+            }
+            return value.containsReference(item);
```

Comment:
I guess this will not work correctly f there are more than one sequence content and should be written somewhat like this:
```
if (value == item || value.containsReference(item)) {
    return true;
}
```